worker/worker.go

```diff
@@ -0,0 +1,113 @@
+package worker
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/bluesky-social/indigo/api/bsky"
+	"github.com/bluesky-social/indigo/xrpc"
+	"github.com/jackc/pgx/v5"
+	"github.com/strideynet/bsky-furry-feed/bluesky"
+	"github.com/strideynet/bsky-furry-feed/store"
+	"go.uber.org/zap"
+)
+
+func Start(
+	ctx context.Context,
+	log *zap.Logger,
+	pdsHost string,
+	bskyCredentials *bluesky.Credentials,
+	pgxStore *store.PGXStore,
+) error {
+	client, err := bluesky.ClientFromCredentials(ctx, pdsHost, bskyCredentials)
+	if err != nil {
+		return err
+	}
+	bgsClient := bluesky.BGSClient{}
+
+	ticker := time.NewTicker(time.Second * 5)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case <-ticker.C:
+			task, err := pgxStore.GetNextFollowTask(ctx)
+			if errors.Is(err, pgx.ErrNoRows) {
+				continue
+			}
+
+			log.Info("processing task", zap.Int64("id", task.ID))
+
+			if task.ShouldUnfollow {
+				err = client.Unfollow(ctx, task.ActorDID)
+			} else {
+				err = client.Follow(ctx, task.ActorDID)
+
+				if err != nil {
+					record, repoRev, err := bgsClient.SyncGetRecord(ctx, "app.bsky.actor.profile", task.ActorDID, "self")
+					if err != nil {
+						if err2 := (&xrpc.Error{}); !errors.As(err, &err2) || err2.StatusCode != 404 {
+							return fmt.Errorf("getting profile: %w", err)
+						}
+						record = nil
+					}
+
+					var profile *bsky.ActorProfile
+					if record != nil {
+						switch record := record.(type) {
+						case *bsky.ActorProfile:
+							profile = record
+						default:
+							return fmt.Errorf("expected *bsky.ActorProfile, got %T", record)
+						}
+					}
+
+					displayName := ""
+					description := ""
+
+					if profile != nil {
+						if profile.DisplayName != nil {
+							displayName = *profile.DisplayName
+						}
+
+						if profile.Description != nil {
+							description = *profile.Description
+						}
+					}
+
+					if err := pgxStore.CreateLatestActorProfile(ctx, store.CreateLatestActorProfileOpts{
```

Comment:
It'd be nice if this was in a transaction with the task success update - but I'm happy to add that as a follow up.